crates/host/src/wasmbus/config.rs

```diff
@@ -1,107 +1,423 @@
+        self.watch_cache
+            .write()
+            .await
+            .insert(name.clone(), rx.clone());
+
+        Ok(ConfigReceiver { name, receiver: rx })
+    }
+}
+
+async fn watcher_loop(
+    store: Store,
+    name: String,
+    tx: watch::Sender<HashMap<String, String>>,
+    done: tokio::sync::oneshot::Sender<anyhow::Result<()>>,
+) {
+    // We need to watch with history so we can get the initial config.
+    let mut watcher = match store.watch_with_history(&name).await {
+        Ok(watcher) => {
+            done.send(Ok(())).expect(
+                "Receiver for watcher setup should not have been dropped. This is programmer error",
+            );
+            watcher
+        }
+        Err(e) => {
+            done.send(Err(anyhow::anyhow!(
+                "Error setting up watcher for {}: {}",
+                name,
+                e
+            )))
+            .expect(
+                "Receiver for watcher setup should not have been dropped. This is programmer error",
+            );
+            return;
+        }
+    };
+    loop {
+        match watcher.try_next().await {
+            Ok(Some(entry)) if matches!(entry.operation, Operation::Delete | Operation::Purge) => {
+                // NOTE(thomastaylor312): We should probably do something and notify something up
+                // the chain if we get a delete or purge event of a config that is still being used.
+                // For now we just zero it out
+                tx.send_replace(HashMap::new());
+            }
+            Ok(Some(entry)) => {
+                let config: HashMap<String, String> = match serde_json::from_slice(&entry.value) {
+                    Ok(config) => config,
+                    Err(e) => {
+                        error!(%name, error = %e, "Error decoding config from store during watch");
+                        continue;
+                    }
+                };
+                tx.send_if_modified(|current| {
+                    if current == &config {
+                        false
+                    } else {
+                        *current = config;
+                        true
+                    }
+                });
+            }
+            Ok(None) => {
+                error!(%name, "Watcher for config has closed");
+                return;
+            }
+            Err(e) => {
+                error!(%name, error = %e, "Error reading from watcher for config. Will wait for next entry");
+                continue;
+            }
+        }
+    }
+}
+
+async fn update_merge(
+    merged_config: &RwLock<HashMap<String, String>>,
+    changed_notifier: &Sender<()>,
+    ordered_receivers: &[Receiver<HashMap<String, String>>],
+) {
+    // We get a write lock to start so nothing else can update the merged config while we merge
+    // in the other configs (e.g. when one of the ordered configs is write locked)
+    let mut lock = merged_config.write().await;
+    lock.clear();
```

Comment:
I can't find this function in the tokio `RwLock` docs, what is this for?

Reply:
This is `clear` from HashMap